benchmarks/benches/rsa.rs

```diff
@@ -0,0 +1,325 @@
+#![allow(non_snake_case)]
+use std::io::Write as _;
+
+use criterion::{criterion_group, criterion_main, BatchSize, Criterion};
+
+use der::Decode as _;
+use pkcs1::DecodeRsaPrivateKey;
+use rand_core::OsRng;
+use ring::signature::KeyPair;
+
+const MSG_STR: &str = "the test message is 32 byte long";
+const MSG: &[u8; 32] = b"the test message is 32 byte long";
+const SALT: &[u8] = b"this is the test salt";
+
+// With the openssl_keygen function above we'd sometimes get keys that are one byte too short.
+// This function sidesteps that issue by rejecting such keys and resampling on rejection.
```

Comment:
That's indeed weird. Would you need to pad it to use it with hacl maybe? But since we don't care too much about RSA right now, let's put it on the pile of things to investigate when we need it.

Reply:
This is due to the way our wrapper code is written - it always assumes that the keys are BITS/8 long. We could loosen that restriction, but (a) at the time I didn't know we would need to loosen it and (b) I am not sure what we'd want to loosen it to. How short is okay? N-1? N-2? Is there a good reason at all for `d` to be shorter? I have opened https://github.com/cryspen/libcrux/issues/841.

Reply:
Yep, RSA is weird. But a byte less would be legal. Anyway, thanks for filing.